src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingSubmissionService.java

```diff
@@ -670,12 +678,36 @@ private String getExerciseTopicForTAAndAbove(long exerciseId) {
         else {
             participations = studentParticipationRepository.findAllWithEagerSubmissionsAndEagerResultsAndEagerAssessorByExerciseId(exerciseId);
         }
-        List<ProgrammingSubmission> submissions = new ArrayList<>();
+        List<ProgrammingSubmission> programmingSubmissions = new ArrayList<>();
         participations.stream().peek(participation -> participation.getExercise().setStudentParticipations(null)).map(StudentParticipation::findLatestLegalOrIllegalSubmission)
                 // filter out non submitted submissions if the flag is set to true
-                .filter(submission -> submission.isPresent() && (!submittedOnly || submission.get().isSubmitted()))
-                .forEach(submission -> submissions.add((ProgrammingSubmission) submission.get()));
-        return submissions;
+                .filter(optionalSubmission -> optionalSubmission.isPresent() && (!submittedOnly || optionalSubmission.get().isSubmitted()))
+                .forEach(optionalSubmission -> programmingSubmissions.add((ProgrammingSubmission) optionalSubmission.get()));
+        return removeExerciseAndSubmissionSet(programmingSubmissions, true);
+    }
+
+    /**
+     * Given a List of ProgrammingSubmissions, this method will remove the attribute participation.exercise.
+     * If removeSubmissionSet = true, also the Set participation.submissions is removed. The number of submissions will be
+     * stored in the attribute participation.SubmissionCount instead of being determined by the size of the set of all submissions.
```

Comment:
```suggestion
     * stored in the attribute participation.submissionCount instead of being determined by the size of the set of all submissions.
```